### PR TITLE
RMI-I2C support

### LINKED ISSUE #17: VoodooI2C Bindings
RMI4 in Linux has I2C bindings alongside with SMBus bindings. This should be very easy to add to this kext as I've already done all the abstractions needed.

VoodooI2C already does have their own Synaptics kext so this is very low priority. I2C bindings for this kext should only be needed if people want to use F12 or the trackpoint.

https://github.com/torvalds/linux/blob/master/drivers/input/rmi4/rmi_i2c.c

### PR DESCRIPTION
**Pull request by zhen-zen**
Earlier PR at https://github.com/VoodooSMBus/VoodooRMI/pull/28
Closes #17

### CODE REVIEW

File: VoodooRMI/Transports/I2C/RMII2C.cpp [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+
+void RMII2C::simulateInterrupt(OSObject* owner, IOTimerEventSource* timer) {
+    interruptOccured(owner, NULL, 0);
+    if (polling)
+        interrupt_simulator->setTimeoutMS(INTERRUPT_SIMULATOR_TIMEOUT);
+}
+
+bool RMII2C::setInterrupt(bool enable) {
+    // interrupt will only be enabled after client update
+    while (!client) {
```
Comment (1Revenger1) [inline VoodooRMI/Transports/I2C/RMII2C.cpp:302]: I'd be very hesistant to use a while loop. handleOpen should already have RMIBus (the client) in it, so I'd maybe use that (it sets a protected variable called `bus`). That said, I don't see it getting used in this function. Maybe could test to see if `bus` is set to Null or not. It'd be preferable if `bus` was used to reference RMIBus.

File: VoodooRMI/Transports/I2C/RMII2C.cpp [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+        IOLog("%s: running on interrupt mode.", getName());
+        setProperty("Interrupt mode", "Native");
+    }
+
+    PMinit();
+    provider->joinPMtree(this);
+    registerPowerDriver(this, RMIPowerStates, 2);
+
+    setProperty("VoodooI2CServices Supported", kOSBooleanTrue);
+    setProperty(HasResetIdentifier, kOSBooleanTrue);
```
Comment (1Revenger1) [inline VoodooRMI/Transports/I2C/RMII2C.cpp:105]: RMII2C does not override reset, so I would not set this property.

Comment (zhen-zen) [reply VoodooRMI/Transports/I2C/RMII2C.cpp:105]: It's an inline one at https://github.com/VoodooSMBus/VoodooRMI/blob/7d43f239f9c6f5c9f3e25d5a429df2b552a551e0/VoodooRMI/Transports/I2C/RMII2C.hpp#L54

Comment (1Revenger1) [reply VoodooRMI/Transports/I2C/RMII2C.cpp:105]: oh, missed that. The RMI4 doesn't specify a reset either so was confused, lol. This is fine

**Comment by 1Revenger1** [comment]
Thanks a lot for putting in the work for this!
Looks good to me

**Pull request merged**